src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityHull.java

```diff
@@ -81,4 +87,30 @@ public void addInformation(ItemStack stack, @Nullable World player, List<String>
         tooltip.add(I18n.format("gregtech.universal.tooltip.voltage_out", energyContainer.getOutputVoltage(), tierName));
         tooltip.add(I18n.format("gregtech.universal.tooltip.energy_storage_capacity", energyContainer.getEnergyCapacity()));
     }
+
+    @Override
+    public void update() {
+        super.update();
+        if (isFirstTick()) {
+            getProxy();
+        }
+    }
+
+    @Nonnull
+    @Override
+    @Optional.Method(modid = GTValues.MODID_APPENG)
+    public AECableType getCableConnectionType(@Nonnull AEPartLocation part) {
+        return AECableType.SMART;
```

Comment:
Should find a way to allow players have the channel gameplay displayed to take advantage of this feature, or they might have to get AE2 stuff.